superset/assets/javascripts/SqlLab/components/SqlEditor.jsx

```diff
@@ -208,7 +149,13 @@ class SqlEditor extends React.PureComponent {
       <div className="sql-toolbar clearfix">
         <div className="pull-left">
           <Form inline>
-            {runButtons}
+            <RunQueryActionButton
+              queryEditor={this.props.queryEditor}
```

Comment:
Curious to hear your thoughts on passing larger objects or more specific props. For instance here we could define a `isTextSelected` prop instead of passing `queryEditor`. Similarly we could pass `allowAsync` instead of `database`.

Isn't it better to pass the more targeted props? It probably means less re-rendering.

I guess it depends on how the component may or may not evolve (say to use other attributes of that object), and whether we'd want to use the component outside of the context of the application / object definition.